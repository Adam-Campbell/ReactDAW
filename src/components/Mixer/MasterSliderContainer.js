import React, { Component } from 'react';
import * as ActionCreators from '../../actions';
import { connect } from 'react-redux';
import MasterSlider from './MasterSlider';

export class MasterSliderContainer extends Component {

    handleVolumeChange = (e) => {
        this.props.setMasterVolume(parseFloat(e.target.value));
    }

    render() {
        return <MasterSlider 
            volume={this.props.volume}
            handleVolumeChange={this.handleVolumeChange}
            trackId="master"
        />
    }
}


const mapStateToProps = state => ({
    volume: state.playerInfo.volume
});

export default connect(
    mapStateToProps,
    {
        setMasterVolume: ActionCreators.setMasterVolume
    }
)(MasterSliderContainer)