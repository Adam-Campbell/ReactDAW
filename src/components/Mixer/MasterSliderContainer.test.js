import React from 'react';
import { shallow } from 'enzyme';
import { MasterSliderContainer } from './MasterSliderContainer';

test('renders correctly', () => {
    const component = shallow(
        <MasterSliderContainer 
            volume={0}
            setMasterVolume={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});