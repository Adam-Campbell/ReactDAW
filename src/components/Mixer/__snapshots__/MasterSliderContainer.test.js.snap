// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MasterSlider
  handleVolumeChange={[Function]}
  trackId="master"
  volume={0}
/>
`;
